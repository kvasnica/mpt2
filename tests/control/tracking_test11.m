function tracking_test11

% bug in mpt_yalmipcftoc rejected tracking=1 for mld systems converted to pwa:
% http://control.ee.ethz.ch/~mpt/hg/mpt-26x?cs=cab227035eee
%
% this also tests fix to a bug in mpt_yalmiptracking when no output constraints
% have been defined (because we used sysStruct.ymin/ymax when creating a
% polytope, but in case of no constraints we set them to +/-Inf:
%
% http://control.ee.ethz.ch/~mpt/hg/mpt-26x?cs=6a515655994c

% MLD system with an equivalent PWA representation
T = evalc('sysStruct = mpt_sys(''pwa_car.hys'');');
probStruct.Q = eye(2); 
probStruct.R = 1; 
probStruct.N = 1; 
probStruct.norm = 2;

probStruct.tracking = 1;
ctrl = mpt_control(sysStruct, probStruct, 'online');
[C, O, V] = mpt_ownmpc(sysStruct, probStruct, 'online');
ctrl2 = mpt_ownmpc(sysStruct, probStruct, C, O, V);

probStruct.tracking = 2;
ctrl = mpt_control(sysStruct, probStruct, 'online');
[C, O, V] = mpt_ownmpc(sysStruct, probStruct, 'online');
ctrl2 = mpt_ownmpc(sysStruct, probStruct, C, O, V);
