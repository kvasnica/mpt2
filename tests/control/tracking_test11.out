Value of "probStruct.subopt_lev" not specified, assuming 0...
Using a PWA model...

The controller will return deltaU=u(k)-u(k-1) instead of u(k) when evaluated!
State vector will have the format x = [x0; u(k-1); reference].
Using a PWA model...

Value of "probStruct.subopt_lev" not specified, assuming 0...
================================================================================
WARNING: offset-free tracking cannot be guaranteed with probStruct.tracking=2 !
================================================================================

Using a PWA model...

================================================================================
WARNING: offset-free tracking cannot be guaranteed with probStruct.tracking=2 !
================================================================================

State vector will have the format x = [x0; reference].
Using a PWA model...
